src/index.ts

```diff
@@ -97,6 +97,37 @@ app.event("reaction_added", async ({ body, client }) => {
   }
 });
 
+if (process.env.SLACK_AUTO_TRANSLATE === "1") {
+  app.message(async ({body, client}) => {
+
+    const event = body.event as MessageEvent;
```

Comment:
inside if/else statements, the type can be a bit more specific. For instance, if you have `if (event.subtype === undefined){ .... }`, the code inside the block can access general message event type objects.

Reply:
I am not sure I follow. The current code translates subtype = {undefined, me_message, message_replied}.
If I just check for undefined I am losing those two subtypes, right?

Reply:
@cbaus I mean that you can do something like this:

```typescript
app.event('message', async ({ event }) => {
  if (event.subtype === undefined) {
    const genericMessageEvent: GenericMessageEvent = event; // this compiles!
    // you can handle new messages
  } else if (event.subtype === 'bot_message') {
    const botMessageEvent: BotMessageEvent = event; // this compiles!
    // you can handle new bot messages
  } else if (event.subtype === 'message_replied') {
    const messageRepliedEvent: MessageRepliedEvent = event; // this compiles!
    // you can handle replied messages
  } else {
    // nothing to do here
  }
});
```

I may not like this style, though.

Reply:
>    const genericMessageEvent: GenericMessageEvent = event; // this compiles!

This line is not necessary in your code. It's just a demonstrating code. You can directly use `event` in a more type-safe way without your own type definition.

Reply:
OK I switched to bolt message type. This is a good comment. I don't need to differentiate between any subtypes as I treat them the same.